lib/block-supports/layout.php

```diff
@@ -66,6 +66,9 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 		$style .= "$selector .alignleft { float: left; margin-right: 2em; margin-left: 0; }";
 		$style .= "$selector .alignright { float: right; margin-left: 2em; margin-right: 0; }";
 		if ( $has_block_gap_support ) {
+			if ( is_array( $gap_value ) ) {
+				$gap_value = $gap_value['left'] === $gap_value['top'] ? $gap_value['top'] : $gap_value['top'] . ' ' . $gap_value['left'];
+			}
 			$gap_style = $gap_value ? $gap_value : 'var( --wp--style--block-gap )';
 			$style    .= "$selector > * { margin-top: 0; margin-bottom: 0; }";
 			$style    .= "$selector > * + * { margin-top: $gap_style;  margin-bottom: 0; }";
```

Comment:
I know you're still exploring here, but just flagging that on this line `$gap_style` needs to be a single value, since it's passed in to `margin-top`.

Reply:
Thanks, totally missed that. 🙇